numba/np/arrayobj.py

```diff
@@ -1989,28 +2039,93 @@ def np_clip(a, a_min, a_max, out=None):
     a_min_is_none = a_min is None or isinstance(a_min, types.NoneType)
     a_max_is_none = a_max is None or isinstance(a_max, types.NoneType)
 
-    def np_clip_impl(a, a_min, a_max, out=None):
-        if a_min_is_none and a_max_is_none:
+    if a_min_is_none and a_max_is_none:
+        # Raises value error when both a_min and a_max are None
+        def np_clip_nn(a, a_min, a_max, out=None):
             raise ValueError("array_clip: must set either max or min")
 
-        ret = np.empty_like(a) if out is None else out
+        return np_clip_nn
+
+    a_min_is_scalar = isinstance(a_min, types.Number)
+    a_max_is_scalar = isinstance(a_max, types.Number)
+
+    if a_min_is_scalar and a_max_is_scalar:
+        def np_clip_ss(a, a_min, a_max, out=None):
+            # a_min and a_max are scalars
+            # since their shape will be empty
+            # broadcast scalars to shape of a
+            # by using np.full_like
+            a_min_full = np.full_like(a, a_min)
+            a_max_full = np.full_like(a, a_max)
+            return _np_clip_impl(a, a_min_full, a_max_full, out)
```

Comment:
Since both `a_min` and `a_max` are scalars, I think it will be faster to compute the result without broadcasting them to arrays.